### PR TITLE
Instagram liker Bot--- Issue 968

### LINKED ISSUE #968: Instagram Liker Bot
# Description

- This script will allow liking of all visible posts by another.
- This script can be used while creating a auto liker site

## Language

- [x] Python

| name             | about                                           | label| assignees    
| ------------- |:-------------:| :-----:|:-----:|
| Instagram Liker Bot      | Will auto like visible instagram post of an user | GSSOC21 , Python | @sayantani11 |

### PR DESCRIPTION
**Pull request by sayantani11**
# Description

Given an username, the bot will search for the account and auto-like all the available post of the user. It will exit by itself after all posts are liked.

Fixes #968 

## Type of change

_Please delete options that are not relevant._
- [x] New feature (non-breaking change which adds functionality)
- [ ] 
# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have created a helpful and easy to understand `README.md`
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by sayantani11** [comment]
@HarshCasper Can you please assist me with the checks?
Like what is the error?

File: Python/Instagram-Liker-Bot/READme.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Use of Script :
```
Comment (vybhav72954) [inline Python/Instagram-Liker-Bot/READme.md:1]: Please follow the [template](https://github.com/HarshCasper/Rotten-Scripts/blob/master/TEMPLATE_README.md). The README needs a lot of alterations.

File: Python/Instagram-Liker-Bot/Instagram_Liker_Bot.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from selenium.webdriver.common.keys import Keys
+from getpass import getpass
+import time
+
+chrome=webdriver.Chrome(ChromeDriverManager().install())
+chrome.get("https://www.instagram.com")
+time.sleep(10)
+
+username=chrome.find_element_by_xpath('//*[@id="loginForm"]/div/div[1]/div/label/input')
+username.send_keys(input("Enter your username")) #enter username
```
Comment (vybhav72954) [inline Python/Instagram-Liker-Bot/Instagram_Liker_Bot.py:12]: Can we make a `.env` file for this input? Would be much safer and standard.

File: Python/Instagram-Liker-Bot/Instagram_Liker_Bot.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+from selenium import webdriver
+from webdriver_manager.chrome import ChromeDriverManager
+from selenium.webdriver.common.keys import Keys
+from getpass import getpass
+import time
+
+chrome=webdriver.Chrome(ChromeDriverManager().install())
```
Comment (vybhav72954) [inline Python/Instagram-Liker-Bot/Instagram_Liker_Bot.py:7]: CAn we have a more modular approach for this? Make a function and call it from a driver?

File: Python/Instagram-Liker-Bot/Instagram_Liker_Bot.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    try:
+        like_button= chrome.find_element_by_xpath('/html/body/div[5]/div[2]/div/article/div[3]/section[1]/span[1]/button')
+        like_button.click()
+        next_button = chrome.find_element_by_xpath('/html/body/div[5]/div[1]/div/div/a[2]')
+        next_button.click()
+        time.sleep(5)
+    except:
+        close_button = chrome.find_element_by_xpath('/html/body/div[5]/div[3]/button')
+        close_button.click()
+        break
```
Comment (vybhav72954) [inline Python/Instagram-Liker-Bot/Instagram_Liker_Bot.py:48]: Missing a blank line at EOF

**Review by vybhav72954 — CHANGES_REQUESTED** [review]
A few changes!

**Comment by sayantani11** [comment]
@vybhav72954  On it....Working on the issue

**Comment by sohamsshah** [comment]
@sayantani11 any updates?

**Comment by sayantani11** [comment]
@sohamsshah I was trying to use the env file and function thing..... Stuck there.... I guess I will able to do it, I am trying

**Comment by devRawnie** [comment]
Hey @sayantani11, You can refer to this for an example of using env files. Be sure to add the .env file in the .gitignore file.
env example

**Comment by vybhav72954** [comment]
@sayantani11 Updates?

**Comment by devRawnie** [comment]
I can work to complete this PR. Kindly assign this to me @vybhav72954

**Comment by vybhav72954** [comment]
> I can work to complete this PR. Kindly assign this to me @vybhav72954

Hey @devRawnie i understand and appreciate your willingness to contribute... But just because someone else is having some problems, doesn't mean I can pass on the work assigned to someone to somebody else entirely without taking consent. Unassigning someone just because they were unable to sort out a small thing would ultimately show my inability as a mentor. You are a frequent contributor and am pretty sure you can take any other issue.

@sayantani11 I will surely help you with the problem by tonight.

**Comment by devRawnie** [comment]
I really appreciate that @vybhav72954. When I asked to be assigned this issue, that came from a place where I just wanted more points in the competition. However, I realized now that this is not just about winning in the long run 😅. More power to you sir 🙌. And I would try and help other developers in any way that I can myself.

File: Python/Instagram-Liker-Bot/Instagram_Liker_Bot.py (hunk 2 above)
Comment (vybhav72954) [reply Python/Instagram-Liker-Bot/Instagram_Liker_Bot.py:12]: Ok so look, here is what you need to do.

```
from decouple import config
```
make one .env file

It should contain your username and password.

Pass those values from the env file. The can be done using `dotenv` as well. Like [here](https://dev.to/biplov/handling-passwords-and-secret-keys-using-environment-variables-2ei0)

Finally there is a `getuser()` functionality as well just like `getpass()` can it be used?

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by sayantani11** [comment]
Okay @vybhav72954 I am looking into it thanks for helping

**Comment by sayantani11** [comment]
@vybhav72954 Thank you for helping with the issue, I was doing it, but unfortunately due to a certain emergency I won't be able to continue with it.
You can assign it to @devRawnie if he wants to work on it.
Although after you helped I was able to proceed.
I am really sorry for this.

**Comment by HarshCasper** [comment]
> @vybhav72954 Thank you for helping with the issue, I was doing it, but unfortunately due to a certain emergency I won't be able to continue with it.
> You can assign it to @devRawnie if he wants to work on it.
> Although after you helped I was able to proceed.
> I am really sorry for this.

No worries! You have put in a lot of hard work in this and we would love to see you succeed. You can continue working on this if you want and we won't have any problems. Let me know what you think.

**Comment by sayantani11** [comment]
@HarshCasper I would love to!!!! Thank You!!! If that's possible I would surely like to continue

**Comment by HarshCasper** [comment]
> @HarshCasper I would love to!!!! Thank You!!! If that's possible I would surely like to continue

The PR would be open for the coming few weeks. Do take a look if you get time or put a comment here so that the stale bot does not close it.

**Pull request closed without merging**